include/mbedtls/net_sockets.h

```diff
@@ -123,6 +123,24 @@ void mbedtls_net_init( mbedtls_net_context *ctx );
  */
 int mbedtls_net_connect( mbedtls_net_context *ctx, const char *host, const char *port, int proto );
 
+/**
+ * \brief          Initiate a connection with host:port in the given protocol with a timeout
+ *
+ * \param ctx      Socket to use
+ * \param host     Host to connect to
+ * \param port     Port to connect to
+ * \param proto    Protocol: MBEDTLS_NET_PROTO_TCP or MBEDTLS_NET_PROTO_UDP
+ * \param timeout  The timeout in milliseconds
+ *
+ * \return         0 if successful, or one of:
+ *                      MBEDTLS_ERR_NET_SOCKET_FAILED,
+ *                      MBEDTLS_ERR_NET_UNKNOWN_HOST,
+ *                      MBEDTLS_ERR_NET_CONNECT_FAILED
+ *
+ * \note           Sets the socket in connected mode even with UDP.
+ */
+int mbedtls_net_connect_timeout( mbedtls_net_context *ctx, const char *host, const char *port, int proto, int timeout );
```

Comment:
Please make this `uint32_t timeout` for compatibility with `mbedtls_net_recv_timeout`. Also, please implement and document that 0 means to wait forever — it's not logical, but I think it's better for the module to be uniform.